app/controllers/sessions_controller.rb

```diff
@@ -50,9 +50,19 @@ def self.create_password_digest(username, password)
 
   def user_from_params
     login = Session.new(session_params)
-    hash = SessionsController.create_password_digest(login.username,
-                                                     login.password)
-    User.where(name: login.username, password: hash).first
+    user = User.find_by(name: login.username)
+    migrate_to_bcrypt(user, login.password) if user.old_password.present?
```

Comment:
Yes. This function will find the user that logged in, and migrate their password to bcrypt without them having to do anything extra. This function makes the decision point around `user.old_password`. If that field is not `nil`, then the code assumes the user's password hasn't been migrated.